Sources/SwiftKafka/New/ClientConfig.swift

```diff
@@ -0,0 +1,278 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// TODO: dict add manually by key (unsafe api)?
+// TODO: support PEMs?
+// TODO: support OAuth?
+// TODO: DocC: inherit documentation?
+// TODO: DocC: take from lirbdkafka official documentation
+// TODO: Topic config -> see KafkaConfig in SwiftKafka
+// TODO: test that values get set accordingly
+// TODO: remove old config tests
+public protocol ClientConfig: Hashable, Equatable {
```

Comment:
I don't think this should be a protocol. We should rather use a struct here and hide the whole dictionary thing internally. Using a protocol for this would require us to either make the consumers generic or use an existential and both are not really great.

Reply:
Furthermore, what I would recommend is that we create a separate `ConsumerConfig` and `ProducerConfig` like you did and then have an internal method on them that can convert them to librdkafka config.